presto-main/src/main/java/com/facebook/presto/execution/scheduler/InternalNodeAndCache.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.scheduler;
+
+import com.facebook.presto.metadata.InternalNode;
+
+public class InternalNodeAndCache
```

Comment:
Hmmmm I'm wondering if we could combine this with `NodeResult` in the previous commit. The content seems to be the same? Also maybe name this `InternalNodeInfo`(I don't have a very good name) in case we want to put more information/context in this class?